ecr.tf

```diff
@@ -13,78 +14,60 @@ resource "aws_ecr_lifecycle_policy" "lifecycle_policy" {
 
   depends_on = [aws_ecr_repository.repos]
 
-  policy = <<EOF
-{
-    "rules": [
-        {
-            "rulePriority": 1,
-            "description": "Keep last ${var.images_to_keep} images with `any` tag",
-            "selection": {
-                "tagStatus": "any",
-                "countType": "imageCountMoreThan",
-                "countNumber": ${var.images_to_keep}
-            },
-            "action": {
-                "type": "expire"
-            }
+  policy = jsonencode({
+    rules = [
+      {
+        rulePriority = 1,
+        description  = "Keep last ${var.images_to_keep} images with `any` tag",
+        selection = {
+          tagStatus   = "any",
+          countType   = "imageCountMoreThan",
+          countNumber = var.images_to_keep
+        },
+        action = {
+          type = "expire"
         }
+      }
     ]
-}
-EOF
+  })
 }
 
-data "template_file" "push_allowed_policy" {
-  template = <<EOF
-{
-    "Sid": "AllowCrossAccountPush",
-    "Effect": "Allow",
-    "Principal": {
-        "AWS": [${join(",", formatlist("\"arn:aws:iam::%s:root\"", var.push_account_list))}]
-    },
-    "Action": [
-        "ecr:PutImage",
-        "ecr:InitiateLayerUpload",
-        "ecr:UploadLayerPart",
-        "ecr:CompleteLayerUpload"
+data "aws_iam_policy_document" "policy_document" {
+  statement {
+    sid    = "AllowCrossAccountPull"
+    effect = "Allow"
+    principals {
+      type        = "AWS"
+      identifiers = local.pull_account_arn_list
+    }
+    actions = [
+      "ecr:GetDownloadUrlForLayer",
+      "ecr:GetRepositoryPolicy",
+      "ecr:BatchGetImage",
+      "ecr:BatchCheckLayerAvailability"
     ]
-}
-EOF
-}
-
-data "template_file" "pull_allowed_policy" {
-  template = <<EOF
-{
-    "Sid": "AllowCrossAccountPull",
-    "Effect": "Allow",
-    "Principal": {
-        "AWS": [${join(",", formatlist("\"arn:aws:iam::%s:root\"", var.pull_account_list))}]
-    },
-    "Action": [
-        "ecr:GetDownloadUrlForLayer",
-        "ecr:GetRepositoryPolicy",
-        "ecr:BatchGetImage",
-        "ecr:BatchCheckLayerAvailability"
```

Comment:
nit: Could put the push first, then pull to minimize the diff here.

Reply:
The reason i put the push first was because when i was testing my ecr-automation for handytrac it was giving me way more diffs due to the order, i think push is second in the original